src/common/warnings.rs

```diff
@@ -0,0 +1,76 @@
+use crate::common::{FileEntry, Warning};
+use std::ffi::OsString;
+use std::fmt;
+
+/// Wraps warnings to provide more information when printing.
+pub struct Warnings {
+    /// Path of the file the warnings originated from.
+    path: FileEntry,
+
+    /// Path of the file's backup.
+    backup_path: Option<OsString>,
+
+    /// List of warnings.
+    pub warnings: Vec<Warning>,
+}
+
+impl Warnings {
+    pub fn new(path: FileEntry, backup_path: Option<OsString>, warnings: Vec<Warning>) -> Self {
+        Self {
+            path,
+            backup_path,
+            warnings,
+        }
+    }
+
+    /// Print warnings without any additional information.
+    pub fn print_warnings(&self) {
+        self.warnings.iter().for_each(|w| println!("{}", w));
+    }
+
+    /// Print the backup file's path.
+    pub fn print_backup(&self) {
+        if let Some(p) = &self.backup_path {
+            println!("Original file was backed up to: {:?}", p);
+        }
+    }
+}
+
+impl fmt::Display for Warnings {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Checking {}", self.path)?;
+        if let Some(p) = &self.backup_path {
+            writeln!(f, "\nOriginal file was backed up to: {:?}", p)?;
```

Comment:
I see the duplicated line https://github.com/dotenv-linter/dotenv-linter/pull/311/files#diff-cefc832d3c36ffe8734f777f9d768fa7R34. Why is this needed?

Reply:
In order to support printing backup files without all the warnings. See below:
```
$ cargo run -- -f -q
Original file was backed up to: ".env_1601943979"
Original file was backed up to: ".env_2/.env_1601943979"

All warnings are fixed. Total: 7
```

Is there a better way of sharing the code to print the backups?

Reply:
I need to think about it 🤔